src/lib/components/Checkbox.react.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+
+import React, { Component } from 'react';
+import MuiCheckbox from '@material-ui/core/Checkbox';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import getMuiTheme from 'material-ui/styles/getMuiTheme';
+
+type Props = {
+  /** Checkbox is checked if true */
+  checked?: boolean,
+  /** CSS class name of the root element */
+  className?: string,
+  /** Mapping from MUI Element name to the intended classname for that Element
+   * See https://material-ui.com/api/checkbox/#css*/
+  classes?: Object,
+  /** Checkbox is disabled if true */
+  disabled?: boolean,
+  /** A callback for firing events to dash */
+  fireEvent?: () => void,
+  /** The element's ID */
+  id: string,
+  /** The label for the checkbox */
+  label: string,
+  /** The name prop of the checkbox */
+  name: string,
+  /** Dash callback to update props on the server */
+  setProps?: () => void,
+  /** Override the inline styles of the root element */
+  style?: Object,
+};
+
+type State = {
+  checked: boolean,
+};
+
+const defaultProps = {
+  checked: false,
+  disabled: false,
+  classes: {},
+  fireEvent: () => {},
+  setProps: () => {},
+  style: {},
+  className: null,
+  name: '',
+  label: null
+};
+
+/** Material UI Checkbox component */
+export default class Checkbox extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {checked: props.checked};
+  }
+
+  UNSAFE_componentWillReceiveProps(nextProps: Props, nextContext: *): void {
+    if (nextProps.checked !== null && nextProps.checked !== this.props.checked) {
+      this.props.setProps({checked: nextProps.checked})
+      // this.setState({checked: nextProps.checked});
```

Comment:
Will the checkbox update correctly from another component (say a button to clear all checks) without this `setState` call?

Reply:
Great catch